examples/jsm/controls/Controls.js

```diff
@@ -0,0 +1,341 @@
+
+		for (var i = 0; i < scope._pointers.length; i++) {
+
+			scope.domElement.releasePointerCapture( scope._pointers[i].pointerId );
+	
+		}
+	
+		scope._pointers.length = 0;
+		scope._keys.length = 0;
+
+	};
+
+	this.dispose = function () {
+
+		scope.enabled = false;
+
+		scope.dispatchEvent( { type: 'dispose' } );
+
+	}
+
+	this.intersectObjects = function( x, y, camera, objects ) {
+
+		var rect = scope.domElement.getBoundingClientRect();
+
+		rayCoord.x = ( x - rect.left ) / rect.width * 2 - 1;
+		rayCoord.y = - ( y - rect.top ) / rect.height * 2 + 1;
+
+		raycaster.setFromCamera( rayCoord, camera );
+
+		intersectedObjects.length = 0;
+
+		raycaster.intersectObjects( objects, true, intersectedObjects );
+
+		return intersectedObjects;
+
+	}
+
+	this.intersectPlane = function ( x, y, camera, plane ) {
+
+		var rect = scope.domElement.getBoundingClientRect();
+
+		rayCoord.x = ( x - rect.left ) / rect.width * 2 - 1;
+		rayCoord.y = - ( y - rect.top ) / rect.height * 2 + 1;
+
+		raycaster.setFromCamera( rayCoord, camera );
+
+		raycaster.ray.intersectPlane( plane, intersectedPoint );
+
+		return intersectedPoint;
+
+	}
+
+	_connect();
+
+}
+
+Controls.prototype = Object.create( EventDispatcher.prototype );
+Controls.prototype.constructor = Controls;
+Controls.prototype.dispatchEvent = function( event ) {
+
+	Object.defineProperty( event, 'target', { writable: true } );
+
+	EventDispatcher.prototype.dispatchEvent.call( this, event );
+
+}
+
+// Pointer class to keep track of pointer movements
+
+var Pointer = function ( pointerEvent ) {
+
+	this.domElement = pointerEvent.srcElement;
+
+	this.pointerId = pointerEvent.pointerId;
+
+	this.type = pointerEvent.pointerType;
+
+	this.button = pointerEvent.button;
+	this.buttons = pointerEvent.buttons;
+
+	this.altKey = pointerEvent.altKey;
+	this.ctrlKey = pointerEvent.ctrlKey;
+	this.metaKey = pointerEvent.metaKey;
+	this.shiftKey = pointerEvent.shiftKey;
+
+	this.start = new Vector2( pointerEvent.clientX, pointerEvent.clientY );
+	this.client = new Vector2( pointerEvent.clientX, pointerEvent.clientY );
+	this.previous = new Vector2( pointerEvent.clientX, pointerEvent.clientY );
+	this.page = new Vector2( pointerEvent.pageX, pointerEvent.pageY );
+	this.movement = new Vector2();
+	this.offset = new Vector2();
+
+	this.update = function ( pointerEvent ) {
+
+		if ( this.pointerId !== pointerEvent.pointerId ) {
+
+			console.error( 'Invalid pointerId!' );
+
+			return;
+
+		}
+
+		this.button = this.button;
+		this.buttons = this.buttons;
+
+		this.ctrlKey = pointerEvent.ctrlKey;
+		this.metaKey = pointerEvent.metaKey;
+		this.shiftKey = pointerEvent.shiftKey;
+
+		this.previous.copy( this.client );
+		this.client.set( pointerEvent.clientX, pointerEvent.clientY );
+		this.page.set( pointerEvent.pageX, pointerEvent.pageY );
+		// Calculate movement because Safari movement is broken
+		this.movement.copy( this.client ).sub( this.previous );
```

Comment:
Controls.js now includes manual movement vector calculation since movementX/Y is broken in Safari.